packages/util/src/u8a/toNumber.ts

```diff
@@ -15,7 +15,11 @@ interface ToNumberOptions {
 export function u8aToNumber (value: Uint8Array, { isNegative = false }: ToNumberOptions = {}): number {
   const count = value.length;
 
-  if (isNegative) {
+  // When the value is a i{8, 16, 24, 32, 40, 40} values and the top-most bit
+  // indicates a signed value, we use a two's complement conversion. If one of these
+  // flags are not set, we just do a normal unsigned conversion (the same shortcut
+  // applies in both the u8aTo{BigInt, Bn} conversions as well)
+  if (isNegative && (value[count - 1] & 0x80)) {
```

Comment:
Is it fine if this goes out of bound?
Also wouldn't we need to check the overall first bit, so basically always `value[0]`?

Reply:
I did this in my branch once as well, keep in mind we need to make a distinction between `positive/negative`, yes. But also between `signed/unsigned` in the first place as well.

We need one more flag or two distinct function for each case.